src/main/java/com/o19s/es/ltr/ranker/parser/XGBoostJsonParser.java

```diff
@@ -153,20 +154,41 @@ private static class SplitParserState {
             PARSER.declareInt(SplitParserState::setNodeId, new ParseField("nodeid"));
             PARSER.declareInt(SplitParserState::setDepth, new ParseField("depth"));
             PARSER.declareString(SplitParserState::setSplit, new ParseField("split"));
-            PARSER.declareFloat(SplitParserState::setThreshold, new ParseField("split_condition"));
             PARSER.declareInt(SplitParserState::setRightNodeId, new ParseField("no"));
             PARSER.declareInt(SplitParserState::setLeftNodeId, new ParseField("yes"));
             PARSER.declareInt(SplitParserState::setMissingNodeId, new ParseField("missing"));
             PARSER.declareFloat(SplitParserState::setLeaf, new ParseField("leaf"));
+            PARSER.declareField((parser, splitParserState, context) -> {
```

Comment:
can we move this to a function?

Reply:
we'll have to create a separate class for it.. as it requires an object extending Parser interface
should we go ahead with that or live with this lambda ?

can create a function which returns this lambda may be

Reply:
have created a separate static class @sabhishekgowda